cmd/upgrade/app/v1alpha2/0.9.0-to-1.0.0/jiva/jiva_upgrade.go

```diff
@@ -0,0 +1,387 @@
+										 "openebs.io/replica": "jiva-replica"
+									 }
+								 },
+					 "topologyKey": "kubernetes.io/hostname"
+							 }
+						 ]
+					 }
+				 }
+			  }
+		   }
+		}
+	 }`
+
+	controllerPatch = `{
+		"metadata": {
+		   "labels": {
+			  "openebs.io/version": "@upgrade_version@"
+		   }
+		},
+		"spec": {
+		   "template": {
+			   "metadata": {
+				   "labels":{
+					   "openebs.io/version": "@upgrade_version@"
+				   }
+			   },
+			  "spec": {
+				 "containers": [
+					{
+					   "name": "@c_name@",
+					   "image": "@controller_image@:@upgrade_version@"
+					},
+					{
+					   "name": "maya-volume-exporter",
+					   "image": "@m_exporter_image@:@upgrade_version@"
+					}
+				 ]
+			  }
+		   }
+		}
+	 }`
+
+	servicePatch = `{
+		"metadata": {
+		   "labels": {
+			  "openebs.io/version": "@upgrade_version@"
+		   }
+		}
+	 }`
+
+	kubeConfigPath = "/home/user/.kube/config"
+
+	deployClient  = deploy.NewKubeClient(deploy.WithKubeConfigPath(kubeConfigPath))
+	serviceClient = svc.NewKubeClient(svc.WithKubeConfigPath(kubeConfigPath))
+	pvClient      = pv.NewKubeClient(pv.WithKubeConfigPath(kubeConfigPath))
+)
+
+func getDeploymentDetails(labels, namespace string) (
+	deployName,
+	containerName,
+	version string,
+	err error,
+) {
+	deployList, err := deployClient.WithNamespace(namespace).List(
+		&metav1.ListOptions{
+			LabelSelector: labels,
+		})
+	if err != nil {
+		return "", "", "", err
+	}
+	if len(deployList.Items) == 1 {
+		deployName = deployList.Items[0].Name
+		version = deployList.Items[0].Labels["openebs.io/version"]
+		if deployName == "" {
+			return "", "", "", errors.New("empty deployment name")
+		}
+		if len(deployList.Items[0].Spec.Template.Spec.Containers) == 0 {
+			return "", "", "", errors.New("empty container list")
+		}
+		containerName = deployList.Items[0].Spec.Template.Spec.Containers[0].Name
+		if containerName == "" {
+			return "", "", "", errors.New("empty container name")
+		}
+	} else {
+		return "", "", "", errors.New("deployment missing")
+	}
+	return deployName, containerName, version, nil
+}
+
+func patchDelpoyment(
+	deployName,
+	namespace string,
+	pt types.PatchType,
+	data []byte,
+) error {
+	var (
+		retries = 60
+	)
+	_, err := deployClient.WithNamespace(namespace).Patch(
+		deployName,
+		pt,
+		data,
+	)
+	if err != nil {
+		return err
+	}
+	for {
+		retries = retries - 1
+		rolloutStatus, err := deployClient.WithNamespace(namespace).
+			RolloutStatus(deployName)
+		if err != nil {
+			return err
+		}
+		time.Sleep(5 * time.Second)
+		if retries == 0 || rolloutStatus.IsRolledout {
+			if !rolloutStatus.IsRolledout {
+				return errors.Errorf("failed to rollout %s", rolloutStatus.Message)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func getBaseImage(deployObj *appsv1.Deployment, name string) (string, error) {
+	for _, con := range deployObj.Spec.Template.Spec.Containers {
+		if con.Name == name {
+			return strings.Split(con.Image, ":")[0], nil
```

Comment:
[optional] Do we have `containers.From(deployObj.Spec.Template.Spec.Containers).getImageForName(name)`